libraries/Microsoft.Bot.Builder.Core/BotContext.cs

```diff
@@ -107,15 +99,20 @@ public bool Responded
                 ApplyConversationReference(a, cr);
             }
 
-            List<Activity> activityList = new List<Activity>(activities);
+            // Convert the IActivities to Activies. 
+            Activity[] activityArray = Array.ConvertAll(activities, (input) => (Activity)input);            
+
+            // Create the list used by the recursive methods. 
+            List<Activity> activityList = new List<Activity>(activityArray);
```

Comment:
at this point you've now got 3 collections of the same objects; `activities`, `activityArray`, and `activityList`. I'm thinking `activities.Select(a=>(Activity)a).ToList()` would be much more efficient.